test/pylib/scylla_cluster.py

```diff
@@ -844,9 +848,33 @@ async def _cluster_server_add(self, _request) -> aiohttp.web.Response:
     async def _cluster_server_remove(self, _request) -> aiohttp.web.Response:
         """Remove a specified server"""
         assert self.cluster
-        server_id = _request.match_info['id']
-        if not await self.cluster.server_remove(server_id):
-            return aiohttp.web.Response(status=500, text=f"Host {server_id} not found")
+        server_ip = _request.match_info['ip']
+        try:
+            await self.cluster.server_remove(server_ip)
+            return aiohttp.web.Response(text="OK")
+        except RuntimeError as exc:
+            return aiohttp.web.Response(status=500, text=f"{exc}")
+
+    async def _cluster_remove_node(self, _request) -> aiohttp.web.Response:
+        """Run remove node on Scylla REST API for a specified server"""
+        assert self.cluster
+        # TODO: only pass UUID
+        to_remove_ip = _request.match_info["ip"]
+        to_remove_uuid = _request.match_info["uuid"]
+        assert len(self.cluster.running) > 2, "Can't remove last running node"
+        initiator_ip = next(s_id for s_id in self.cluster.running if s_id != to_remove_ip)
+        logging.info("_cluster_remove_node initiator %s server %s %s", initiator_ip,
+                     to_remove_ip, to_remove_uuid)
+
+        # initate remove
+        try:
+            await self.api.remove_node_and_wait(initiator_ip, to_remove_uuid)
```

Comment:
Pass `timeout` here.
Take it from the request.
On the client side, extend the API to take the timeout and send it.
In the test send the timeout.

Reply:
Wait methods were not needed so removed in https://github.com/scylladb/scylladb/commit/7cc3aedb8d083a86de5cf4d92fd4c3190af4413d